### PR TITLE
drivers: npmx: shell: NTC module control based on ntc type

### PR DESCRIPTION
**Pull request by jaz1-nordic**
JIRA: NRFX-4131

### CODE REVIEW

**Review by annakielar — APPROVED** [review]

File: drivers/npmx/npmx_shell.c [hunk 1]
```diff
@@ -1700,9 +1726,43 @@ static int cmd_adc_ntc_set(const struct shell *shell, size_t argc, char **argv,
...
 	npmx_adc_ntc_type_t type = (npmx_adc_ntc_type_t)data;
 
+	if (modules_mask & NPMX_CHARGER_MODULE_NTC_LIMITS_MASK) {
+		if (type == NPMX_ADC_NTC_TYPE_HI_Z) {
+			err_code = npmx_charger_module_disable_set(
+				charger_instance, NPMX_CHARGER_MODULE_NTC_LIMITS_MASK);
+			if (check_error_code(shell, err_code)) {
+				shell_info(
+					shell,
+					"Info: The NTC temperature limit control module has been disabled.");
```
Comment (annakielar) [inline drivers/npmx/npmx_shell.c:1738]:
```suggestion
					"Info: the NTC temperature limit control module has been disabled.");
```

**Review by annakielar — CHANGES_REQUESTED** [review]
As mentioned offline, please check if indents for shell_error and shell_info are in line with npmx standards.

File: drivers/npmx/npmx_shell.c [hunk 2]
```diff
@@ -401,12 +401,37 @@ static int cmd_charger_module_set(const struct shell *shell, size_t argc, char *
...
+	npmx_error_t err_code;
+
+	if ((charger_set == 1) && (mask == NPMX_CHARGER_MODULE_NTC_LIMITS_MASK)) {
+		npmx_adc_ntc_type_t type;
+		npmx_adc_t *adc_instance = npmx_adc_get(npmx_instance, 0);
+
+		err_code = npmx_adc_ntc_get(adc_instance, &type);
+
+		if (check_error_code(shell, err_code)) {
+			if (type == NPMX_ADC_NTC_TYPE_HI_Z) {
```
Comment (wilu-nordic) [inline drivers/npmx/npmx_shell.c:413]: It is not required to check whether NTC temperature limit control module was enabled after setting thermistor.

Comment (jaz1-nordic) [reply drivers/npmx/npmx_shell.c:413]: This is the case when the thermistor type is not configured (hi_z) and the user wants to enable the ntc module.
So, in my opinion, it is absolutely necessary.

Comment (wilu-nordic) [reply drivers/npmx/npmx_shell.c:413]: Shell is not a place where those checks should be done. For example in BUCK, selecting a pin to enable buck, the pin should be checked whether is it selected as input. We decided not to add that logic to the shell, the user is responsible for the proper command order.

Comment (jaz1-nordic) [reply drivers/npmx/npmx_shell.c]: I don't understand why this is not a good place? If something is not allowed to work, it is good practice to inform the user about a possible error.
Anyway, done as you wanted.

**Review by wilu-nordic — APPROVED** [review]

**Review by annakielar — APPROVED** [review]

**Pull request merged**